elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/SQLQueryEngine.java

```diff
@@ -304,21 +305,34 @@ private String extractJoin(Path.PathElement pathElement) {
         Class<?> entityClass = pathElement.getType();
         String entityAlias = FilterPredicate.getTypeAlias(entityClass);
 
-        Table tableAnnotation = dictionary.getAnnotation(relationshipClass, Table.class);
-
-        String relationshipTableName = (tableAnnotation == null)
-                ? dictionary.getJsonAliasFor(relationshipClass)
-                : tableAnnotation.name();
 
         String relationshipIdField = getColumnName(relationshipClass, dictionary.getIdFieldName(relationshipClass));
-
-        return String.format("LEFT JOIN %s AS %s ON %s.%s = %s.%s",
-                relationshipTableName,
-                relationshipAlias,
-                entityAlias,
-                getColumnName(entityClass, relationshipName),
-                relationshipAlias,
-                relationshipIdField);
+        String relationshipColumnName = getColumnName(entityClass, relationshipName);
+        Subselect subselectAnnotation = dictionary.getAnnotation(relationshipClass, Subselect.class);
+
+        if (subselectAnnotation == null) {
+            Table tableAnnotation = dictionary.getAnnotation(relationshipClass, Table.class);
+
+            String relationshipTableName = (tableAnnotation == null)
+                    ? dictionary.getJsonAliasFor(relationshipClass)
+                    : tableAnnotation.name();
+
+            return String.format("LEFT JOIN %s AS %s ON %s.%s = %s.%s",
```

Comment:
The SQL fragment is so similar - can we combine them and just change the tableName variable?

Reply:
👍